cmd/snap/cmd_run.go

```diff
@@ -313,6 +313,9 @@ func createOrUpdateUserDataSymlink(info *snap.Info, usr *user.User) error {
 }
 
 func createUserDataDirs(info *snap.Info) error {
+	oldUmask := syscall.Umask(0)
```

Comment:
afaiu  we also need to use runtime.(Un)LockOSThread around this.

Reply:
Hmm, on second though, umask is a process property. Can you justify why this may be necessary?

Reply:
You are right, I probably read the man page at some point too quickly, sometimes they mix up processes with threads. It also means though that a go program can change this only very early unless if are very sure it doesn't do anything interesting fs-wise in goroutines.

Reply:
I don't know if it means we need a comment here